### PR TITLE
adjustments regarding the new subgroup behavior since keycloak 23.0.0

### PR DESCRIPTION
**Pull request by leon-manq**
Hi,
frist of all nice libaray. Thank you for all the great work.
Since Keycloak 23.0.0 there is a new endpoint which is used to provide the subgroups of a group. 
GET /admin/realms/{realm}/groups/{id}/children
Retrieving information from the subgroup property is no longer possible. The array is empty.

### CODE REVIEW

File: src/Representation/Group.php [hunk 1]
```diff
@@ -43,6 +44,9 @@ public function __construct(
         protected ?array $realmRoles = null,
         #[Since('23.0.0')]
         protected ?int $subGroupCount = null,
+        #[Since('23.0.0')]
+        protected ?string $parentId = null,
```
Comment (fschmtt) [inline src/Representation/Group.php:48]: Please add a getter and setter to the class's annotation (adhere to the order: getters first, setters second, alphabetical order).

**Comment by leon-manq** [comment]
Thanks for the review. I have implemented the points you mentioned. PHP is not my primary language. It took a moment to understand the implementation of the getters and setters with reflection.

File: src/Resource/Groups.php [hunk 2]
```diff
@@ -41,7 +56,7 @@ public function get(string $realm, string $groupId): Group
         );
     }
 
-    public function create(string $realm, Group $group): void
+    public function create(string $realm, Group $group, ?string $parentGroupId = null): void
```
Comment (fschmtt) [inline src/Resource/Groups.php:59]: You forgot to remove the `$parentGroupId` parameter 🙂

**Review by fschmtt — CHANGES_REQUESTED** [review]

**Comment by fschmtt** [comment]
The current (integration) testing setup runs **all** tests against each Keycloak version.

Since the child group endpoints exist only in >= 23.0.0 we need to skip those tests when the version tested against is lower.

In #83 I added the possibility to skip tests depending on the Keycloak version.

In your case the test could look like this:
```php
public function testCreateChildGroup(): void
{
    $this->skipIfKeycloakVersionIsLessThan('23.0.0');

    $importedGroupName = Uuid::uuid4()->toString();
    $childGroupName = Uuid::uuid4()->toString();

    $groups = $this->getKeycloak()->groups();

    // Create group
    $groups->create('master', new Group(name: $importedGroupName),);
    $group = $groups->all('master')->first();
    static::assertInstanceOf(Group::class, $group);

    // Create child group
    $groups->createChild('master', new Group(name: $childGroupName), $group->getId());
    $childGroups = $groups->children('master', $group->getId());
    static::assertCount(1, $childGroups);

    $childGroup = $childGroups->first();
    static::assertInstanceOf(Group::class, $childGroup);
    static::assertSame($childGroupName, $childGroup->getName());
}
```

So, what you need to do:
* Rebase your branch to `main` 
* Adjust the `Groups` test (see suggestion above)

**Review by fschmtt — APPROVED** [review]

**Comment by fschmtt** [comment]
Thanks for your contribution, I appreciate it! 🥳

The changes are included in the v0.17.0 release.

**Pull request merged**